### PR TITLE
[codegen/python] Handle ResourceType

### LINKED ISSUE #5411: [codegen/python] Handle ResourceType
Depends on #5325

### PR DESCRIPTION
**Pull request by lblackstone**
Fix #5411

### CODE REVIEW

**Comment by lblackstone** [comment]
Here's the current generated code: https://gist.github.com/lblackstone/f94a6a309907c1df1308604cb5281ce4

File: pkg/codegen/python/gen.go [hunk 1]
```diff
@@ -1845,26 +1928,46 @@ func generateModuleContextMap(tool string, pkg *schema.Package, info PackageInfo
...
 
 	inputSeen, outputSeen := codegen.Set{}, codegen.Set{}
-	visitObjectTypesFromProperties(pkg.Config, outputSeen, func(t *schema.ObjectType) {
-		getModFromToken(t.Token).details(t).outputType = true
+	visitObjectTypesFromProperties(pkg.Config, outputSeen, func(t interface{}) {
+		switch T := t.(type) {
+		case *schema.ObjectType:
+			getModFromToken(T.Token).details(T).outputType = true
+		case *schema.ResourceType:
+			getModFromToken(T.Token)
```
Comment (justinvp) [inline pkg/codegen/python/gen.go:1936]: I don't think we need the `case *schema.ResourceType: getModFromToken(T.Token)` in any of the calls to `visitObjectTypes`/`visitObjectTypesFromProperties` inside `generateModuleContextMap`. Did you find there was a need for it?

File: pkg/codegen/python/gen.go [hunk 2]
```diff
@@ -510,16 +552,26 @@ func (mod *modContext) genTypes(dir string, fs fs) error {
...
 			if input && mod.details(t).inputType {
-				visitObjectTypesFromProperties(t.Properties, inputSeen, func(t *schema.ObjectType) {
-					imports.addIf(mod, t.Token, true /*input*/, func(imp string) bool {
-						// No need to import `._inputs` inside _inputs.py.
-						return imp != "from ._inputs import *"
-					})
+				visitObjectTypesFromProperties(t.Properties, inputSeen, func(t interface{}) {
+					switch T := t.(type) {
+					case *schema.ObjectType:
+						imports.addIf(mod, T.Token, true /*input*/, func(imp string) bool {
```
Comment (justinvp) [inline pkg/codegen/python/gen.go:558]: Should we rename `add` and `addIf` to `addType` and `addTypeIf`, now that we have `addResource`?

File: pkg/codegen/python/gen.go [hunk 3]
```diff
@@ -460,13 +485,30 @@ func (mod *modContext) importFromToken(tok string, input bool) string {
 	return fmt.Sprintf("from %s import %[2]s as _%[2]s", relImport, components[0])
 }
 
-// emitConfigVariables emits all config vaiables in the given module, returning the resulting file.
+func (mod *modContext) importResourceFromToken(tok string) string {
```
Comment (justinvp) [inline pkg/codegen/python/gen.go:488]: Now that we have `importResourceFromToken`, should we rename `importFromToken` to `importTypeFromToken`?

**Review by justinvp — APPROVED** [review]
LGTM otherwise

File: pkg/codegen/python/gen.go (hunk 3 above)
Comment (lblackstone) [reply pkg/codegen/python/gen.go:488]: Sounds reasonable

File: pkg/codegen/python/gen.go (hunk 1 above)
Comment (lblackstone) [reply pkg/codegen/python/gen.go:1936]: I updated all of the callsites for `visitObjectTypesFromProperties`, but didn't check to see if they were all necessary.

Comment (lblackstone) [reply pkg/codegen/python/gen.go:1936]: Looks like those weren't needed, so removed them.

**Pull request merged**